Sources/SourceKit/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+if(CMAKE_VERSION VERSION_LESS 3.16)
+  if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+    set(CMAKE_LINK_LIBRARY_FLAG "-l")
+    set(CMAKE_LINK_LIBRARY_SUFFIX "")
+  endif()
+endif()
+
+add_library(SourceKit
+  DocumentManager.swift
+  SourceKitLSPCommandMetadata.swift
+  SourceKitServer+Options.swift
+  SourceKitServer.swift
+  ToolchainLanguageServer.swift
+  Workspace.swift)
+target_sources(SourceKit PRIVATE
+  clangd/ClangLanguageServer.swift)
+target_sources(SourceKit PRIVATE
+  sourcekitd/CommentXML.swift
+  sourcekitd/CursorInfo.swift
+  sourcekitd/Diagnostic.swift
+  sourcekitd/EditorPlaceholder.swift
+  sourcekitd/SemanticRefactorCommand.swift
+  sourcekitd/SemanticRefactoring.swift
+  sourcekitd/SwiftCommand.swift
+  sourcekitd/SwiftLanguageServer.swift
+  sourcekitd/SwiftSourceKitFramework.swift)
+set_target_properties(SourceKit PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
+# TODO(compnerd) reduce the exposure here, why is everything PUBLIC-ly linked?
+target_link_libraries(SourceKit PUBLIC
+  Csourcekitd
+  BuildServerProtocol
+  IndexStoreDB
+  LanguageServerProtocol
+  LanguageServerProtocolJSONRPC
+  SKCore
+  SKSwiftPMWorkspace
+  TSCUtility)
+
+if(BUILD_SHARED_LIBS)
```

Comment:
Do we need all these shared libs? The main product is the executable. We added a library product for the LanguageServerProtocol module, but not for the others.

Reply:
Yes, sadly we do need the shared libraries because on Windows, we do not support static libraries.

Reply:
Can't we just combine everything  - or at least everything outside of LangaugeServerProtocol module - into the single executable?

Reply:
As long as we can do that in the s-p-m based build as well (basically collapse the directory structure), yes, that would be possible.

Reply:
Oh I get it now. I was thinking of what we did in indexstoredb, but that was C++ code. With Swift code we can’t combine into one target because we need to keep the module structure.